src/security/src/main/kotlin/org/vaccineimpact/api/security/OneTimeTokenAuthenticator.kt

```diff
@@ -0,0 +1,36 @@
+package org.vaccineimpact.api.security
+
+import com.nimbusds.jwt.JWT
+import org.pac4j.core.credentials.TokenCredentials
+import org.pac4j.core.exception.CredentialsException
+
+class OneTimeTokenAuthenticator(
+        tokenHelper: WebTokenHelper,
+        private val oneTimeTokenChecker: OneTimeTokenChecker
+) : MontaguTokenAuthenticator(tokenHelper, TokenType.ONETIME)
+{
+    override fun createJwtProfile(credentials: TokenCredentials, jwt: JWT)
+    {
+        super.createJwtProfile(credentials, jwt)
+        checkTokenAgainstRepository(credentials)
+    }
+
+    private fun checkTokenAgainstRepository(credentials: TokenCredentials)
+    {
+        if (!oneTimeTokenChecker.checkToken(credentials.token))
+        {
+            throw CredentialsException("Token has already been used (or never existed)")
+        }
+    }
+
+    override fun handleUrlAttribute(credentials: TokenCredentials, jwt: JWT)
+    {
+        val claims = jwt.jwtClaimsSet
+        val url = claims.getClaim("url")
+        @Suppress("UselessCallOnNotNull")
```

Comment:
I see this is the same as the reporting API, but shall we just rewrite in a way that doesn't cause warnings? Looks like the `if` condition is equivalent to:
```
url == null || url.toString().isEmpty()
```

Reply:
How's this version (https://github.com/vimc/montagu-api/pull/242/commits/5ba6f201185e25831558bb962f54f48eb1be964a); I think it's clearer. The important thing is that we handle the case (without throwing a weird exception) where the URL claim is missing, or a date, or something strange like that.